blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineRunImpl.java

```diff
@@ -107,6 +108,22 @@ public BlueRunState getStateObj() {
         } catch (InterruptedException | TimeoutException e) {
             logger.error("Error getting StateObject from execution context: "+e.getMessage(), e);
         }
+
+        // TODO: Probably move this elsewhere - maybe into PipelineNodeContainerImpl?
+        boolean isQueued = false;
+        boolean isRunning = false;
+        for (BluePipelineNode n : getNodes()) {
+            if (n.getStateObj().equals(BlueRunState.QUEUED)) {
+                isQueued = true;
+            } else if (n.getStateObj().equals(BlueRunState.RUNNING)) {
+                isRunning = true;
+            }
+        }
+        if (isQueued && !isRunning) {
+            return BlueRunState.QUEUED;
+
+        }
+
```

Comment:
@abayer you are seeing a `toLowerCase()` issue because something here is now returning null and this method is `@Nonnull`